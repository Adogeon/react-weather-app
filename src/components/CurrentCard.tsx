import styles from "./CurrentCard.module.css";

interface CurrentCardProps {
  data: {
    condition: {
      icon: string;
      text: string;
    };
    temp_c: string;
    temp_f: string;
    feelslike_c: string;
    feelslike_f: string;
  };
}

const CurrentCard = (props: CurrentCardProps) => {
  return (
    <section id="current" className={styles.current}>
      <div className={styles["icon-container"]}>
        <img
          src={props.data.condition.icon}
          alt={`${props.data.condition.text}-icon`}
        />
      </div>
      <div className={styles["weather-info"]}>
        <div>{props.data.condition.text}</div>
        <div className={styles.temperature}>{props.data.temp_c}°C</div>
        <div className={styles["feels-like"]}>
          Feels like {props.data.feelslike_c}°C
        </div>
      </div>
    </section>
  );
};

export default CurrentCard;
