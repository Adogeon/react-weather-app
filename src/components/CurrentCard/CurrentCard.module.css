.current {
  flex: 0 1 auto;
  display: flex;
  justify-content: space-around;
  flex-wrap:wrap;
  align-items: center;
  background-color: #2e2e2e;
  padding: 20px;
  border-radius: 8px;
  border: 2px solid #cccccc;
}

.icon-container {
  flex: 1;
}

.icon-container img {
  width: 7.5rem;
  height: 7.5rem;
}

.weather-info {
  flex: 2;
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.weather-info div {
  font-size: 1.2rem;
  color: #f0f0f0;
}

.temperature {
  font-weight: bold;
  font-size: 1.5rem;
  color: #ffcc00;
}

.feels-like {
  font-size: 1rem;
  color: #cccccc;
}
