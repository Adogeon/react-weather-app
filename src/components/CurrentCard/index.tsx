import styles from "./CurrentCard.module.css";
import type { CurrentWeather } from "../../types/weather";

const CurrentCard = ({ data }: { data: CurrentWeather }) => {
  const bigIconSrc = data.condition.icon.replace(/\/64x64\//, "/128x128/");
  return (
    <section id="current" className={styles.current}>
      <div className={styles["icon-container"]}>
        <img src={bigIconSrc} alt={`${data.condition.text}-icon`} />
      </div>
      <div className={styles["weather-info"]}>
        <div>{data.condition.text}</div>
        <div className={styles.temperature}>{data.temp_c}°C</div>
        <div className={styles["feels-like"]}>
          Feels like {data.feelslike_c}°C
        </div>
      </div>
    </section>
  );
};

export default CurrentCard;
