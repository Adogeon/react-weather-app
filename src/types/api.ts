import type { CurrentWeather, DailyForecastWithHour } from "./weather";

export interface rawAPIError {
    error: {
        code: string;
        message: string;
    };
}

export interface rawAPISuccess {
    location: {
        name: string;
        region: string;
        country: string;
    };
    current: CurrentWeather
    forecast: {
        forecastday: Array<DailyForecastWithHour>
    }
};
