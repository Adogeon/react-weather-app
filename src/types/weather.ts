export interface WeatherCondition {
    text: string;
    icon: string;
}

export interface TempAndConditions {
    temp_c: string;
    temp_f: string;
    feelslike_c: string;
    feelslike_f: string;
    condition: WeatherCondition;
}

export interface CurrentWeather extends TempAndConditions {
    last_updated: string
}

interface DailyForecastWeather {
    maxtemp_c: string;
    maxtemp_f: string;
    mintemp_c: string;
    mintemp_f: string;
    daily_chance_of_rain: string;
    avghumidity: string;
    condition: WeatherCondition;
}

export interface DailyForecastData {
    date: string;
    day: DailyForecastWeather
}

interface HourlyForecast extends TempAndConditions {
    time: string;
}

export interface DailyForecastWithHour extends DailyForecastData {
    hour: Array<HourlyForecast>
}