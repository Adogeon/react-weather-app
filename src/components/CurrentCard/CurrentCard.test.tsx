import { render } from "@testing-library/react";
import { test, expect } from "vitest";

import CurrentCard from ".";
import { CurrentWeather } from "../../types/weather";

const testData: CurrentWeather = {
  condition: {
    text: "test text",
    icon: "/64x64/test",
  },
  temp_c: "20",
  temp_f: "70",
  feelslike_c: "10",
  feelslike_f: "50",
  last_updated: "12329812",
};

test("The CurrentCard should render correctly", () => {
  const { asFragment } = render(<CurrentCard data={testData} />);

  expect(asFragment()).toMatchSnapshot();
});
