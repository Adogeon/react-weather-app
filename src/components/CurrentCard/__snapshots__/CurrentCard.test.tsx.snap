// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`The CurrentCard should render correctly 1`] = `
<DocumentFragment>
  <section
    class="_current_8601b2"
    id="current"
  >
    <div
      class="_icon-container_8601b2"
    >
      <img
        alt="test text-icon"
        src="/128x128/test"
      />
    </div>
    <div
      class="_weather-info_8601b2"
    >
      <div>
        test text
      </div>
      <div
        class="_temperature_8601b2"
      >
        20°C
      </div>
      <div
        class="_feels-like_8601b2"
      >
        Feels like 10°C
      </div>
    </div>
  </section>
</DocumentFragment>
`;
